package ifg;

// Representa um modelo ou grupo de modelos. 
message Modelo {
  optional string id = 1;  // id do modelo de acordo com arquivo de modelos.asciiproto
  optional string texto = 2;  // texto do menu, se houver. Caso contrario usa id.
  optional int32 peso = 3;  // para grupos aleatorios, indica o peso do modelo (% = peso / total de pesos).

  // Caso o modelo seja de um grupo, modelos nao sera vazio.
  repeated Modelo modelos = 4; // modelos que compoe o grupo. 
  optional bool aleatorio = 5; // se verdadeiro, que o grupo sera gerado aleatoriamente.
  optional string quantidade = 6; // quantidade de membros gerados, por exemplo: 1d4+1.
}

// O menu de modelos pode conter um submenu ou modelos.
message MenuModelos {
  optional string id = 1;
  repeated MenuModelos sub_menu = 2;
  repeated Modelo modelo = 3;
}
